inc/updater/namespace.php

```diff
@@ -55,25 +104,55 @@ function upgrader_pre_download() : bool {
  * @return array
  */
 function add_accept_header( $args, $url ) : array {
-	global $metadata, $release;
+	global $release;
 
-	$accept_header = [];
 	if ( ! str_contains( $url, 'api.github.com' ) ) {
 		return $args;
 	}
-	foreach ( $release->artifacts->package[0] as $key => $value ) {
-		$key = str_replace( '-', '_', $key );
-		$package[ $key ] = $value;
-	}
-	if ( isset( $package['content_type'] ) && $package['content_type'] === 'application/octet-stream' ) {
-		if ( ! empty( $accept_header ) && str_contains( $url, $metadata->slug ) ) {
-			$args = array_merge( $args, $accept_header );
+
+	foreach ( $release as $rel ) {
+		if ( $url === $rel->artifacts->package[0]->url ) {
+			$content_type = $rel->artifacts->package[0]->{'content-type'};
+			if ( $content_type === 'application/octet-stream' ) {
+				$args = array_merge( $args, [ 'headers' => [ 'Accept' => $content_type ] ] );
+			}
 		}
 	}
 
 	return $args;
 }
 
+/**
+ * Get data from DID.
+ *
+ * @param  string $id DID.
+ *
+ * @return mixed
+ */
+function get_release_from_did( $id ) {
```

Comment:
```suggestion
function get_latest_release_from_did( $id ) {
```
Since this will only ever return the latest release, might as well be specific here.

Reply:
You need to fix the call too.